src/Main.ps1

```diff
@@ -907,7 +907,7 @@ function Invoke-Pester {
 
             if ('Diagnostic' -eq $PesterPreference.Output.Verbosity.Value) {
                 $PesterPreference.Debug.WriteDebugMessages = $true
-                $PesterPreference.Debug.WriteDebugMessagesFrom = "Discovery", "Skip", "Filter", "Mock", "CodeCoverage"
+                $PesterPreference.Debug.WriteDebugMessagesFrom = "Discovery", "Skip", "Mock", "CodeCoverage"
             }
```

Comment:
Is it right that Diagnostic should always override the debug-categories? These are already set as default in PesterConfiguration-object. If I use advanced configuration with Diagnostic output and have manually added the 'Filter' category, this will remove that.

Maybe we should rely completely on the default in PesterConfiguration? Or at least just ensure here that all these are present **as a minimum**?

Reply:
I think you can check on the object if it was changed and only set this if not changed. I probably did not take that in consideration, because adding the Diagnostic output was an afterthought :)

Reply:
If `IsOriginalValue() -eq $true` then it would already have those categories defined (set in `DebugConfiguration.cs`). No need to re-apply here. What about WriteDebugMessages itself? We should probably keep force-enabled here like it is, right? Can't remember if Diagnostic provides any value to the user without any debugmessages enabled.

Reply:
It does not, it is Detailed view + Debug log. I checked the code, we should probably just append the values, they are array of strings, and we don't have to care if there are duplicates. That way you will get what user specified in config or pesterpreference (or both, we merge the object above this call) plus the values from Diagnostic.

Reply:
Ended up adding the missing values only to keep the configuration clean. First to avoid the discussed leak in advanced config (now fixed), but then realized it's visible in passthru object.